File: 1470. Shuffle the Array/1470. Shuffle the Array.c
/**
 * Note: The returned array must be malloced, assume caller calls free().
 */
int* shuffle(int* nums, int numsSize, int n, int* returnSize){
    int* result=(int*) malloc(numsSize* sizeof(int));
    int mid=numsSize/2;
    for(int i=0; i<mid;i++){
            result[2*i]=nums[i];
            result[2*i+1]=nums[mid+i];        
    }
    *returnSize=numsSize;
    return result;
}

File: 1470. Shuffle the Array/1470. Shuffle the Array.java
class Solution {
    public int[] shuffle(int[] nums, int n) {
        int len=nums.length;
        int mid=len/2;
        int result[] = new int[len];
        for(int i=0; i<mid; i++){
            result[2*i]=nums[i];
            result[2*i+1]=nums[mid+i];
        }
        return result;
    }
}
